samples/FsSync.js

```diff
@@ -0,0 +1,37 @@
+var fs = require('fs');
+
+console.log('File System sample');
+
+var fd = fs.openSync('testfile.txt', 'w+');
+
+var stats = fs.statSync('testfile.txt');
+
+if (stats.isFile()) {
+    console.log('fd is file');
+} else if (stats.isDirectory()) {
+    console.log('fd is directory');
+} else {
+    console.log('fd has unknown type');
+}
+
+var wbuf = new Buffer('write some bytes');
+
+var written = fs.writeSync(fd, wbuf, 0, wbuf.length, 0);
+
+console.log('wrote ' + written + ' bytes');
+
+
+var rbuf = new Buffer(16);
+var ret = fs.readSync(fd, rbuf, 0, 16, 0);
+console.log('read ' + ret + ' bytes from file');
+console.log('buffer=' + rbuf.toString('ascii'));
+
+var fd1 = fs.openSync('file.txt', 'w');
+
+fs.writeFileSync('string.txt', 'string data', {});
+fs.writeFileSync('buffer.txt', new Buffer('buffer data'));
+
+var files = fs.readdirSync('/');
+for (var i = 0; i < files.length; ++i) {
+    console.log('files[' + i + ']=' + files[i]);
+}
```

Comment:
It is user error, but for ashell we can't just let user errors break things.

Actually this points out, the cleanup functions aren't really needed in "production mode" (vs. ashell). So we would save some space by #ifdef'ing out the cleanup stuff in that case. In this case where we might have to keep a list of open file descriptors - that would be another thing that could be #ifdef'd.

How about at least put a TODO comment in the file that we should clean up leaked file descriptors someday.